README.md

```diff
@@ -42,87 +42,76 @@ Find a list of all the available options for your DogStatsD Client in the [DogSt
 ### Migrating from v4.x to v5.x
 
 If you are already using DogStatsD-ruby v4.x and you want to migrate to a version v5.x, the major
-change concerning you is the new threading model (please see section Threading model):
+change concerning you is the new [threading model](#threading-model):
 
 In practice, it means two things:
 
-1. Now that the client is buffering metrics before sending them, you have to manually
-call the method `Datadog::Statsd#flush` if you want to force the sending of metrics. Note that the companion thread will automatically flush the buffered metrics if the buffer gets full or when you are closing the instance.
+1. Now that the client is buffering metrics before sending them, you have to call `Datadog::Statsd#flush(sync: true)` if you want synchronous behavior. In most cases, this is not needed, as the companion thread will automatically flush the buffered metrics if the buffer gets full or when you are closing the instance.
 
 2. You have to make sure you are either:
 
-  * using singletons instances of the DogStatsD client and not allocating one each time you need one, letting the buffering mechanism flush metrics, it's still a bad solution if the process later forks (see related section below). Or,
-  * properly closing your DogStatsD client instance when it is not needed anymore using the method `Datadog::Statsd#close` to release the resources used by the instance and to close the socket
+  * Using a singleton instance of the DogStatsD client instead of creating a new instance whenever you need one; this will let the buffering mechanism flush metrics regularly
```

Comment:
I think the note about `fork` here is still appropriate

Reply:
I thought https://github.com/DataDog/dogstatsd-ruby/pull/199 was already in, but indeed it's not yet. Since I'll be out for a few days anyway, I propose that we wait to merge this PR until that other one is merged.

Reply:
Makes sense - our main dev for this library (@remeh) is out until Monday anyways so it would be good for him to take a look as well.

Reply:
Note for myself: mergeable once 5.3.0 live.